packages/components/src/dynamic-form/dynamic-form.tsx

```diff
@@ -50,21 +61,13 @@ export const DynamicForm: React.FC< DynamicFormProps > = ( {
 	const fields =
 		baseFields instanceof Array ? baseFields : Object.values( baseFields );
 
-	const getInitialConfigValues = () =>
-		fields.reduce(
-			( data, field ) => ( {
-				...data,
-				[ field.id ]:
-					field.type === 'checkbox'
-						? field.value === 'yes'
-						: field.value,
-			} ),
-			{}
-		);
+	const initialValues = useMemo( () => getInitialConfigValues( fields ), [
+		fields,
+	] );
```

Comment:
This wasn't required, but thought it would be nice to have.

Reply:
Nice! 💯